src/components/App/index.tsx

```diff
@@ -83,18 +87,32 @@ export default class App extends Component<Props, State> {
   }
 
   @bind
-  private onBack() {
-    this.setState({ file: undefined });
+  private onPopState() {
+    this.setState({ isEditorOpen: history.pathname === ROUTE_EDITOR });
+  }
+
+  @bind
+  private openEditor() {
+    history.push(ROUTE_EDITOR);
+    this.setState({ isEditorOpen: true });
+  }
+
+  componentDidMount() {
+    history.addPopStateListener(this.onPopState);
+  }
+
+  componentWillUnmount() {
+    history.removePopStateListener(this.onPopState);
```

Comment:
When does the main app unmount?

Reply:
Right now, only on full page refresh so this might not be needed 😄